### DISCUSSION TITLE
dynamic routes

### DISCUSSION THREAD

**Discussion by imadatyatalah** [vercel/next.js#20206] [Help] [answered]
i want to create 4 pages with different API calls (i don't want to create for each API call a page) any solution?

**Comment by stigkj** [comment, marked as answer]
Do you want to call different APIs depending on the dynamic part of your path? If so, you could do something like this:

The page is `pages/post/[pid].js` and has the following content:

```
import { useRouter } from 'next/router'

const Post = () => {
  const router = useRouter()
  const { pid } = router.query

  if( pid === '1' ) {
    const data = <API call 1>
    return <p>Data for 1: {data}</p>
  } else if( pid === '2' ) {
    const data = <API call 2>
    return <p>Data for 2: {data}</p>
  } else if 
}

export default Post
```
